packages/transaction-controller/src/gas-flows/LineaGasFeeFlow.ts

```diff
@@ -0,0 +1,209 @@
+    } catch (error) {
+      log('Using default flow as fallback due to error', error);
+      return new DefaultGasFeeFlow().getGasFees(request);
+    }
+  }
+
+  async #getLineaGasFees(request: GasFeeFlowRequest) {
+    const { ethQuery, getGasFeeControllerEstimates, transactionMeta } = request;
+
+    const lineaResponse = await this.#getLineaResponse(
+      transactionMeta,
+      ethQuery,
+    );
+
+    log('Received Linea response', lineaResponse);
+
+    const gasFeeControllerEstimates = await getGasFeeControllerEstimates();
+
+    log('Received gas fee controller estimates', gasFeeControllerEstimates);
+
+    if (
+      gasFeeControllerEstimates.gasEstimateType !==
+      GAS_ESTIMATE_TYPES.FEE_MARKET
+    ) {
+      throw new Error('No gas fee estimates available');
+    }
+
+    const baseFees = this.#getBaseFees(lineaResponse);
+
+    const priorityFees = this.#getPriorityFees(
+      lineaResponse,
+      gasFeeControllerEstimates.gasFeeEstimates,
+    );
+
+    const maxFees = this.#getMaxFees(baseFees, priorityFees);
+
+    this.#logDifferencesToGasFeeController(
+      maxFees,
+      gasFeeControllerEstimates.gasFeeEstimates,
+    );
+
+    return {
+      estimates: {
+        low: {
+          maxFeePerGas: toHex(maxFees.low),
+          maxPriorityFeePerGas: toHex(priorityFees.low),
+        },
+        medium: {
+          maxFeePerGas: toHex(maxFees.medium),
+          maxPriorityFeePerGas: toHex(priorityFees.medium),
+        },
+        high: {
+          maxFeePerGas: toHex(maxFees.high),
+          maxPriorityFeePerGas: toHex(priorityFees.high),
+        },
+      },
+    };
+  }
+
+  #getLineaResponse(
+    transactionMeta: TransactionMeta,
+    ethQuery: EthQuery,
+  ): Promise<LineaEstimateGasResponse> {
+    return query(ethQuery, 'linea_estimateGas', [
+      {
+        from: transactionMeta.txParams.from,
+        to: transactionMeta.txParams.to,
+        value: transactionMeta.txParams.value,
+        input: transactionMeta.txParams.data,
+        gasPrice: '0x100000000',
+      },
+    ]);
+  }
+
+  #getBaseFees(lineaResponse: LineaEstimateGasResponse): FeesByLevel {
+    const baseFeeLow = hexToBN(lineaResponse.baseFeePerGas);
+    const baseFeeMedium = baseFeeLow.muln(BASE_FEE_MULTIPLIERS.medium);
+    const baseFeeHigh = baseFeeLow.muln(BASE_FEE_MULTIPLIERS.high);
+
+    return {
+      low: baseFeeLow,
+      medium: baseFeeMedium,
+      high: baseFeeHigh,
+    };
+  }
+
+  #getPriorityFees(
+    lineaResponse: LineaEstimateGasResponse,
+    gasFeeEstimates: GasFeeEstimates,
+  ): FeesByLevel {
+    const mediumPriorityIncrease = gweiDecToWEIBN(
+      gasFeeEstimates.medium.suggestedMaxPriorityFeePerGas,
+    ).sub(gweiDecToWEIBN(gasFeeEstimates.low.suggestedMaxPriorityFeePerGas));
+
+    const highPriorityIncrease = gweiDecToWEIBN(
+      gasFeeEstimates.high.suggestedMaxPriorityFeePerGas,
+    ).sub(gweiDecToWEIBN(gasFeeEstimates.medium.suggestedMaxPriorityFeePerGas));
```

Comment:
Could we have a helper function that takes two priorities and a `GasFeeEstimates` and computes the difference between the `suggestedMaxPriorityFeePerGas` of the two levels?

Or a helper that takes two gweiDec's and computes the difference as wei in BN?

Reply:
Done.